# Data loading

This notebook loads the Letterboxd ratings (partial) dataset and inspects its structure. 
Initial steps include loading the dataset, inspecting its structure, and computing basic statistics. 

The goal of this notebook is to upload and clean the dataset of movie ratings on the website Letterboxd from the most popular users. 

The dataset is taken from Titouan Mokrani, Letterboxd Movies Rating 10M (2025), available on Hugging Face.

## Dataset overview

In [ ]:
import pandas as pd

In [ ]:
ratings = pd.read_parquet("../data/raw/ratings.parquet")

In [ ]:
ratings.info()

<class 'pandas.DataFrame'>
RangeIndex: 10462257 entries, 0 to 10462256
Data columns (total 4 columns):
 #   Column    Dtype  
---  ------    -----  
 0   user      uint16 
 1   movie_id  uint32 
 2   title     str    
 3   rating    float32
dtypes: float32(1), str(1), uint16(1), uint32(1)
memory usage: 345.5 MB


In [ ]:
ratings.head(5)

,user,movie_id,title,rating
0,0,1045381,scream-7,0.5
1,0,1090728,ella-mccay,4.0
2,0,977342,zootopia-2,3.5
3,0,871148,wicked-for-good,4.0
4,0,358757,now-you-see-me-now-you-dont-2025,4.0


In [ ]:
ratings.shape

(10462257, 4)

In [ ]:
ratings.columns

Index(['user', 'movie_id', 'title', 'rating'], dtype='str')

In [ ]:
ratings.dtypes

user         uint16
movie_id     uint32
title           str
rating      float32
dtype: object

In [ ]:
ratings["rating"].describe()

count    1.046226e+07
mean     3.283513e+00
std      1.055683e+00
min      5.000000e-01
25%      2.500000e+00
50%      3.500000e+00
75%      4.000000e+00
max      5.000000e+00
Name: rating, dtype: float64

In [ ]:
ratings["user"].nunique()

6519

In [ ]:
ratings["movie_id"].nunique()

281045

The dataset contains ratings in the website Letterboxd of 6519 users of 281045 movies. 

There are over 10 million ratings recorded. This is a subset of the site but from the most popular users. 

For each rating we have information of the user, the name and id of the movie and a rating from 0.5 to 5.0. 

The dataset is clean and complete enough to start taking a deeper look on it. 